src/foolscap/banana.py

```diff
@@ -1,7 +1,7 @@
 
 import six
 import struct, time
-
+from past.builtins import long
```

Comment:
I suppose this is a negligible change compared to the py3k porting integration branch (since this is merely replacing the six.PY3 conditional that sets long to int, below)... however...

A lot of the existing int vs long code in this module is questionable at best.  It was written when int and long were more distinct than they are in Python 2.7.  The further change on Python 3 to make int and long actually be two names for the same type makes that questionable code just plain silly.

For example:

```
            elif typebyte == INT:
                obj = int(header)
```

`header` is already an int.  This is a confusing no-op.

And worse:

```
            elif typebyte == NEG:
                # -2**31 is too large for a positive int, so go through
                # LongType first
                obj = int(-long(header))
```
`header` is already an int and now `int` is `long`.  It's not entirely clear what the point of this code is anyway.  To force values small enough to fit into the int type to be represented as an int, I suppose?  But I don't have any idea what the *point* of that is.  The two types are basically interchangeable and any code that breaks because -2 ** 31 gets represented as a long instead of as an int is in for a lot of trouble regardless of what Foolscap does.

Also, since everyone is on 64 bit now, the critical threshold isn't even 2**31 anymore - it's 2**63.

Reply:
Yup. Lots of such code in a few places in banana.py . Let me take a fresh look but a bit conflicted as to whether try these changes for now or when I port the test for this module.

Reply:
I would rather see no `long` in the code at all, so the sooner we get there the better :)

Reply:
Ok.